stl/inc/ranges

```diff
@@ -728,71 +734,340 @@ namespace ranges {
         using all_t = decltype(all(_STD declval<_Rng>()));
     } // namespace views
 
+    // CLASS TEMPLATE ranges::filter_view
+    // clang-format off
+    template <input_range _Vw, indirect_unary_predicate<iterator_t<_Vw>> _Pr>
+        requires view<_Vw> && is_object_v<_Pr>
+    class filter_view : public _Cached_position_t<forward_range<_Vw>, _Vw, filter_view<_Vw, _Pr>> {
+        // clang-format on
+    private:
+        /* [[no_unique_address]] */ _Vw _Range{};
+        /* [[no_unique_address]] */ _Semiregular_box<_Pr> _Pred{};
+
+        template <class Traits> // TRANSITION, LWG-3289
+        struct _Iterator_base {};
+        // clang-format off
+        template <class Traits>
+            requires _Has_member_iterator_category<Traits>
+        struct _Iterator_base<Traits> {
+            // clang-format on
+            using iterator_category =
+                conditional_t<derived_from<typename Traits::iterator_category, bidirectional_iterator_tag>,
+                    bidirectional_iterator_tag,
+                    conditional_t<derived_from<typename Traits::iterator_category, forward_iterator_tag>,
+                        forward_iterator_tag, input_iterator_tag>>;
+        };
+
+        class _Iterator : public _Iterator_base<iterator_traits<iterator_t<_Vw>>> {
+        public:
+            using iterator_concept = conditional_t<bidirectional_range<_Vw>, bidirectional_iterator_tag,
+                conditional_t<forward_range<_Vw>, forward_iterator_tag, input_iterator_tag>>;
+            using value_type       = range_value_t<_Vw>;
+            using difference_type  = range_difference_t<_Vw>;
+
+            _Iterator() = default;
+            constexpr _Iterator(filter_view& _Parent_, iterator_t<_Vw> _Current_) noexcept(
+                is_nothrow_move_constructible_v<iterator_t<_Vw>>) // strengthened
+                : _Current(_STD move(_Current_)), _Parent{_STD addressof(_Parent_)} {}

```

Comment:
I believe we should be verifying that `_Current_` is within `_Parent->_Range`. Otherwise, any precondition on `_Current` can be cirumvented.

Reply:
I'll `_Adl_verify_range(_Current, _RANGES end(_Parent_)._Range)` here for IDL2.